lib/client.ts

```diff
@@ -55,28 +57,40 @@ class Client {
     }
 
 
-    send(requestCommand: string, requestData: any, destinationTwinId: number, expirationMinutes: number) {
-        // create new envelope with given data and destination
-        const envelope = new Envelope({
-            uid: uuidv4(),
-            timestamp: Math.round(Date.now() / 1000),
-            expiration: expirationMinutes * 60,
-            source: this.source,
-            destination: new Address({ twin: destinationTwinId })
-        });
-        if (requestCommand) {
-            envelope.request = new Request({ command: requestCommand })
-        }
-        if (requestData) {
-            envelope.plain = new Uint8Array(Buffer.from(requestData));
+    async send(requestCommand: string, requestData: any, destinationTwinId: number, expirationMinutes: number) {
+        try {
+            // create new envelope with given data and destination
+            const envelope = new Envelope({
+                uid: uuidv4(),
+                timestamp: Math.round(Date.now() / 1000),
+                expiration: expirationMinutes * 60,
+                source: this.source,
+
+            });
+            // need to check if destination twinId exists by fetching dest twin from chain first
+            this.destTwin = await getTwinFromTwinID(destinationTwinId, this.chainUrl)
+
+            envelope.destination = new Address({ twin: this.destTwin.id })
+
+            if (requestCommand) {
+                envelope.request = new Request({ command: requestCommand })
+            }
+            if (requestData) {
+                envelope.plain = new Uint8Array(Buffer.from(requestData));
+
+            }
+            const clientEnvelope = new ClientEnvelope(this.signer, envelope, this.chainUrl);
+            // send enevelope binary using socket
+            this.con.send(clientEnvelope.serializeBinary());
+            // add request id to responses map on client object
+            this.responses.set(clientEnvelope.uid, clientEnvelope)
+            return clientEnvelope.uid;
+
+        } catch (err) {
+            this.con.close();
```

Comment:
resources cleanup needs to happen in the finally

Reply:
no need since the connection needs to be closed only in case of error